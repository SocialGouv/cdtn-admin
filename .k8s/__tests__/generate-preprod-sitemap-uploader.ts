//

import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

jest.setTimeout(1000 * 60);
test("kosko generate --preprod jobs/sitemap-uploader", async () => {
  expect(
    await getEnvManifests("preprod", "jobs/sitemap-uploader", {
      ...project("cdtn-admin").preprod,
    })
  ).toMatchSnapshot();
});
