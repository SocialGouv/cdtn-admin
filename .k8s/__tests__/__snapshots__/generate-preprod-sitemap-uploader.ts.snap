// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kosko generate --preprod jobs/sitemap-uploader 1`] = `
"---
apiVersion: batch/v1
kind: Job
metadata:
  annotations:
    app.gitlab.com/app: socialgouv-cdtn-admin
    app.gitlab.com/env: preprod-dev2
    app.gitlab.com/env.name: preprod-dev2
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    kapp.k14s.io/update-strategy: fallback-on-replace
  name: sitemap-uploader
spec:
  backoffLimit: 3
  template:
    metadata:
      name: sitemap-uploader
      annotations:
        kapp.k14s.io/deploy-logs: for-new-or-existing
    spec:
      restartPolicy: OnFailure
      containers:
        - name: az-sitemap-uploader
          image: mcr.microsoft.com/azure-cli:2.9.1
          command:
            - bash
          args:
            - '-c'
            - |+

              echo \\"Fetch sitemap from $SITEMAP_ENDPOINT\\"

              curl --fail -L $SITEMAP_ENDPOINT -o sitemap.xml

              if [[ -f sitemap.xml ]]; then
                # replace the default urls hostname if $BASE_URL is given
                [[ -n $BASE_URL ]] && sed -i -E 's#<loc>https?://[^/]*/#<loc>'$BASE_URL'/#' sitemap.xml
                # upload
                echo \\"Upload sitemap to azure/$DESTINATION_CONTAINER/$DESTINATION_NAME\\"
                az storage blob upload --account-name $AZ_ACCOUNT_NAME --account-key $AZ_ACCOUNT_KEY --container-name $DESTINATION_CONTAINER --file sitemap.xml --name $DESTINATION_NAME
              else
                echo \\"Cannot fetch sitemap.xml, abort\\"
                exit 1
              fi

          env:
            - name: BASE_URL
              value: https://cdtn-preprod-code-travail.dev2.fabrique.social.gouv.fr
            - name: DESTINATION_CONTAINER
              value: sitemap
            - name: DESTINATION_NAME
              value: sitemap-preprod.xml
            - name: SITEMAP_ENDPOINT
              value: >-
                https://preprod-cdtn-admin.dev2.fabrique.social.gouv.fr/api/sitemap
          envFrom:
            - secretRef:
                name: sitemap-uploader
---
apiVersion: bitnami.com/v1alpha1
kind: SealedSecret
metadata:
  annotations:
    sealedsecrets.bitnami.com/cluster-wide: 'true'
  name: sitemap-uploader
spec:
  encryptedData:
    AZ_ACCOUNT_KEY: >-
      AgAnrmuh4U4qK+Q19Zwr9RI1YJuGDfGzAy3GE/hMzWuoykHh3IdFFlRHWCtXvFThO0yVkSNWqboqVGMIpXlnUYAeSM0iDTtKDwrlktm0fxLW9TIm11I7qqSJHXzKW6ZEuPPpTKm30jdAd7aA4KFyFF9n5v0i99BRqs+YurEbSskfTLNIQlfXOuq0LToNdTQniQoGKtuvSzkhdXM1JYnxSdZaQV6nM6j3X1WHlxEIqWdgSBd/EHCCvRPyjZe2Dvvd40epuMLwFwNM+1+4W2TP40AjyqCQvPwxXDa5V7rbX70JmieO738/fwbP97g4N5I6Q/tpJZwtyi1Sk0Cf/kKn7/NqGBKU01TAgxMBCw350Edv6JXxYzRlqDTUi9f8XtIFNrNAmTWe7zQFPxCNKVw9CHfNdUTKWG5VU5F9wMy98KOg5t75OPFU8F04ppViZA5f0pEY50LG9A/c/HjTSgPYKpTDXs88cvrWz7w6Oc0fAXN88iJqGYDo2dx3VJ1OpJBinpQwXrkuHqpQTMB1DcWppkPMsWPxhIouAwcHozdt7ap8z6X9hIW7qwHQcuX0OuDPDczv7SqK8JygK68P4mj4sZqdaHenT14o8Pn3v8w5WKSfOnNNyebKhvlHGduVR2N4ye1zY5l9meKKbylbpG6CP1IpKlzO84ru8/nFVTQSaHdDTAYoWm3hVBPp6chkCRCC/Pba05RYvBqkfJVwwDG+DAudmkmZKFNOOnqWV2NA4klfUISvPkXgewzkfYwBsIjuiN8A/xcJVcn1qZSONRl5YDuJe21U2DGOdzoUyF3KNMdvoiYlj9kOlixQ
    AZ_ACCOUNT_NAME: >-
      AgBlhiAia0DAc/mF2yX1Ea7c0D6E9YE4Nv/Ejvhv775Tc2Giu9gymt+FV2Zi2JGiS5Ubo6A1PKgxFGHUVHR5src3Ncg5Jsrkeape2iJjxNF17mf6oderADKDjqR5Ad8BuskiWjAocC5lfMh7QskmxM2Bmzp53UiTxgp88qsriRrM77LB6At3aGlUqAsPt5C9QJgJ7QXE0TPasGFr0MBJieSeKb2dVfBrOX9280bM6u4ckMKLuhTThtdavtZX4kcx7ZRCJjsTFZJiXoDTYj74/2sTiR1fRTHfhAXQu6IaeQPvFWK8gPlIEgtLI8PdvmjoLm+2oT7+k6DEbfJNUDraW05rWH1NcaegEjYqHPzMALfvkFSmtDbyIRO0HTv2LSGWkxy3J8rXP9kfGFnkqjbzLXFxxME7DLFTjVeV95biY3s266gUZKN0wPmWXFDCAvTGRQgf9dODRT4rLc8NhruCZQ3Ma5mp9JRlUkCVS+1by2bIDd8m1Uro5HstR7KUaxMLry//QUdSRs6hjWt7hGLGyxr5hd7YDO3E85kENyOelkB3ixUAPYvQHbimyt/EpoorgvIPorQenRHMV6Hsix7tHQNpuX7jGhpIh6HZ7p/L+nojvo550odjuenFrDryFduehxnrMIq/qnp3jPRtcbyP2hfjoeIAymu3H7d+LHOUfwVDd7++T44jE9Or9SEtnRr3oYMvJiXWoSet/Z3se3k=
  template:
    metadata:
      annotations:
        sealedsecrets.bitnami.com/cluster-wide: 'true'
      name: sitemap-uploader
    type: Opaque
"
`;
